datamapper/repo.py

```diff
@@ -47,20 +48,29 @@ def __init__(self, database: Database):
         sql = func.count().select().select_from(sql)
         return await self.database.fetch_val(sql)
 
-    async def insert(self, model: Type[Model], **values: Any) -> Model:
-        sql_values = collect_sql_values(model, values)
+    async def insert(
+        self, model_or_changeset: Union[Type[Model], Changeset[Model]], **values: Any
```

Comment:
I think this should probably be:

 ```suggestion
        def insert(self, record_or_changeset: Union[Model, Changeset[Model]]) -> Union[Model, Changeset]:
```

If you get a `record` (which is just an instance of a model), we'll insert it's `record.attributes`. That's how Ecto does it. That way, we don't have to worry about a scenario where we get a changeset and additional values.